Documentation/Usage/Components.rst

```diff
@@ -0,0 +1,257 @@
+    <f:argument name="icon" type="string" optional="{true}" />
+
+    <button class="myButton myButton--{variant}">
+        <f:if condition="{icon}">
+            <my:atom.icon name="{icon}" />
+        </f:if>
+        <f:slot />
+    </button>
+
+..  note::
+    IDE autocomplete for all available components via XSD files, similar to ViewHelpers,
+    is not implemented yet, but is planned for a future release.
+
+.. _components-context:
+
+Providing Context
+=================
+
+Sometimes it might be helpful to provide some global settings to all components
+within one component collection. One common use case could be to provide design
+tokens from a JSON file to your components.
+
+The :php:`AbstractComponentCollection` provides the `getAdditionalVariables()`,
+which allows you to do just that:
+
+..  code-block:: php
+    namespace Vendor\MyPackage\Components;
+
+    use TYPO3Fluid\Fluid\Core\Component\AbstractComponentCollection;
+
+    final class ComponentCollection extends AbstractComponentCollection
+    {
+        private ?array $designTokens = null;
+
+        // ...
+
+        public function getAdditionalVariables(string $viewHelperName): array
+        {
+            $this->designTokens ??= json_decode(file_get_contents('path/to/designTokens.json'), true);
+            return [
+                'designTokens' => $this->designTokens,
+            ];
+        }
+    }
+
+In your component templates you would then be able to access those tokens:
+
+..  code-block:: xml
+    <f:argument name="color" type="string" optional="{true}" default="brand" />
+
+    <div style="background-color: {designTokens.colors.{color}}"></div>
+
+.. _components-arbitrary-arguments:
+
+Allowing Arbitrary Arguments
+============================
+
+By default, components only accept arguments that are defined explicitly via
+the :ref:`<f:argument> ViewHelper <typo3fluid-fluid-argument>`. However, there might
+be use cases where you would like to accept arbitrary arguments.
+
+This is possible by defining :php:`additionalArgumentsAllowed()` in your
+:php:`ComponentCollection` implementation (in this example for all components
+in the collection):
+
+..  code-block:: php
+    namespace Vendor\MyPackage\Components;
+
+    use TYPO3Fluid\Fluid\Core\Component\AbstractComponentCollection;
+
+    final class ComponentCollection extends AbstractComponentCollection
+    {
+        // ...
+
+        protected function additionalArgumentsAllowed(string $viewHelperName): bool
+        {
+            return true;
+        }
+    }
+
+The following call of the button component would then be valid:
+
+..  code-block:: xml
+    {namespace my=Vendor\MyPackage\Components\ComponentCollection}
+
+    <my:atom.button something="my text">
+        Button label
+    </my:atom.button>
+
+In the component, `{something}` would be available as an additional variable.
+
+.. _components-folder-structure:
+
+Alternative Folder Structure
```

Comment:
This could be mentioned/linked in the introduction part since this structure is what would expect based on existing Fluid conventions.

Reply:
I've cross-referenced it.